sdk/python/feast/telemetry.py

```diff
@@ -30,29 +32,38 @@
 
 class Telemetry:
     def __init__(self):
-        feast_home_dir = join(expanduser("~"), ".feast")
-        Path(feast_home_dir).mkdir(exist_ok=True)
-        telemetry_filepath = join(feast_home_dir, "telemetry")
-        self._telemetry_enabled = (
+        self._telemetry_enabled = False
+        self.check_env()
+
+    def check_env(self):
+        telemetry_enabled = (
             os.getenv("FEAST_TELEMETRY", default="True") == "True"
         )  # written this way to turn the env var string into a boolean
 
-        self._is_test = os.getenv("FEAST_IS_TELEMETRY_TEST", "False") == "True"
-
-        if self._telemetry_enabled:
-            self._telemetry_counter = {"get_online_features": 0}
-            if os.path.exists(telemetry_filepath):
-                with open(telemetry_filepath, "r") as f:
-                    self._telemetry_id = f.read()
-            else:
-                self._telemetry_id = str(uuid.uuid4())
-
-                with open(telemetry_filepath, "w") as f:
-                    f.write(self._telemetry_id)
-                print(
-                    "Feast is an open source project that collects anonymized usage statistics. To opt out or learn"
-                    " more see https://docs.feast.dev/v/master/feast-on-kubernetes/advanced-1/telemetry"
-                )
+        # Check if it changed
+        if telemetry_enabled != self._telemetry_enabled:
+            self._telemetry_enabled = telemetry_enabled
+
+            if self._telemetry_enabled:
+                feast_home_dir = join(expanduser("~"), ".feast")
+                Path(feast_home_dir).mkdir(exist_ok=True)
+                telemetry_filepath = join(feast_home_dir, "telemetry")
+
+                self._is_test = os.getenv("FEAST_IS_TELEMETRY_TEST", "False") == "True"
+                self._telemetry_counter = {"get_online_features": 0}
+
+                if os.path.exists(telemetry_filepath):
+                    with open(telemetry_filepath, "r") as f:
+                        self._telemetry_id = f.read()
+                else:
+                    self._telemetry_id = str(uuid.uuid4())
+
+                    with open(telemetry_filepath, "w") as f:
+                        f.write(self._telemetry_id)
+                    print(
+                        "Feast is an open source project that collects anonymized error reporting and usage statistics. To opt out or learn"
+                        " more see https://docs.feast.dev/v/master/feast-on-kubernetes/advanced-1/telemetry"
```

Comment:
Same here, I think we should create a nicer URL. We also shouldnt point to master.